index.js

```diff
@@ -157,10 +164,19 @@ class InlineSourceWebpackPlugin {
           }
         });
       }
-      compiler.hooks.emit.tapAsync(name, (compilation, callback) => {
-        this._deleteAsset(compilation);
-        callback && callback();
-      });
+      if (Compilation && Compilation.PROCESS_ASSETS_STAGE_SUMMARIZE) {
+        compiler.hooks.thisCompilation.tap(name, compilation => {
+          compilation.hooks.processAssets.tap(
+            { name, stage: Compilation.PROCESS_ASSETS_STAGE_SUMMARIZE },
```

Comment:
There are a number of stages at which to hook into the pre-sealing asset processing phase; https://github.com/webpack/webpack/blob/d009b13aa231957936c394bb2cf7df275709193a/lib/Compilation.js#L4440-L4519

The one that seemed most appropriate was `PROCESS_ASSETS_STAGE_OPTIMIZE_INLINE`.
However, that phase runs *before* the `beforeEmit` hook fires. The `beforeEmit` / `_process` work is necessary to build up the list of assets-to-delete.

Instead, I used `PROCESS_ASSETS_STAGE_SUMMARIZE` which runs *after* the `beforeEmit` hook fires, and therefore has access to the final list of assets-to-delete.